### DISCUSSION TITLE
Console.ReadLine - Float (ponto vs virgula)

### DISCUSSION THREAD

**Discussion by Dallonso** [balta-io/2801#36] [Perguntas e Respostas] [answered]
Boa tarde Teacher, tudo certo?

Gostaria de saber se há algum método para que o usuário possa entrar com valores decimais usando tanto ponto, quanto virgula?
Em meu código da calculadora, quando digito algum valor decimal usando ponto, o mesmo é ignorado ao executar Console.WriteLine().

Imagem do problema:
![image](https://user-images.githubusercontent.com/15056126/152652154-0940bfae-522e-4116-ab29-5adba41d2431.png)

Se o usuário digitar virgula, tudo corre normalmente:
![image](https://user-images.githubusercontent.com/15056126/152652248-c576e8b2-efa4-4f2a-baba-00b9fc66ee68.png)

Abaixo meu cod:
```
Console.Clear();

Console.WriteLine("Primeiro Valor: ");
float value1 = float.Parse(Console.ReadLine());

Console.WriteLine("Segundo Valor: ");
float value2 = float.Parse(Console.ReadLine());

float result= value1 + value2;

Console.WriteLine($"O valor da soma é de {result}");

```

Grato!

**Comment by niziul** [comment, marked as answer]
Olha, não tenho muita certeza, mas vou me arriscar á dizer que esse problema é algo relacionando com sintaxe de entrada de argumentos no console. 

Minha memória tá um pouco fraca, mas lembro de ter aprendido que quando digitamos números flutuantes no console utilizamos a vírgula ao invés do ponto. Porém, na programação nada é impossível de se resolver.

Abaixo vou deixar uma solução que eu montei para identificar quando há o ponto no input, na verdade, tá mais para gabiarra!

```
float primeiroValorCorreto, segundoValorCorreto;

Console.Clear();

Console.WriteLine("Primeiro Valor: ");
string primeiroValor = Console.ReadLine();

Console.WriteLine("Segundo Valor: ");
string segundoValor = Console.ReadLine();

primeiroValorCorreto = primeiroValor.Contains(".") ? Convert.ToSingle(primeiroValor) / 10 : Convert.ToSingle(primeiroValor);
segundoValorCorreto = segundoValor.Contains(".") ? Convert.ToSingle(segundoValor) / 10 : Convert.ToSingle(segundoValor);

float soma = primeiroValorCorreto + segundoValorCorreto;

Console.WriteLine($"O valor da soma é de {soma.ToString("F")}");
```

+ Resultado da aplicação:
![](https://i.ibb.co/p3RxPFm/imagem-2022-02-06-225257.png)

Aliás, após alguns testes, o código acima só funciona nas dezenas, ou seja, gabiarra.

**Reply by andrebaltieri** [reply]
Muito obrigado pela ajuda @NiziulLuizin! É uma forma de resolver o problema sim!

@Dallonso não se preocupe tanto com isto no momento. O console é mais para apresentações básicas, no ASP.NET você verá mais sobre culturas de interface.

**Reply by Dallonso** [reply]
Opa, vlw pela explicação @NiziulLuizin.  

@andrebaltieri Maravilha. Ansioso para evoluir no curso. Vlw o feedback.

**Reply by Dallonso** [reply]
@andrebaltieri 
@NiziulLuizin 

Na aula de strings conheci o .Replace e acabei inserindo ele no código da Calculadora. Ficou assim:

`float value1 = float.Parse(Console.ReadLine().Replace(".",","));`

Com isso acabou dando certo o input tanto com ponto como com virgula.

![image](https://user-images.githubusercontent.com/15056126/153515925-6f19785d-0243-4eed-9dc7-eebc16cab5dd.png)

![image](https://user-images.githubusercontent.com/15056126/153515988-75bf92de-05b5-4cef-9786-ce6643e2c8b2.png)

@andrebaltieri Minha duvida agora é: Essa seria uma boa solução?

Grato
